lib/flutter_barcode_listener.dart

```diff
@@ -78,15 +79,30 @@ class _BarcodeKeyboardListenerState extends State<BarcodeKeyboardListener> {
     //remove any pending characters older than bufferDuration value
     checkPendingCharCodesToClear();
     _lastScannedCharCodeTime = DateTime.now();
+    callWithoutLineFeed();
     if (char == lineFeed) {
+      _lineFeedCall = true;
       _onBarcodeScannedCallback.call(_scannedChars.join());
       resetScannedCharCodes();
     } else {
       //add character to list of scanned characters;
       _scannedChars.add(char!);
     }
   }
-
+  void callWithoutLineFeed() {
+    if (_timer?.isActive ?? false) {
+      _timer?.cancel();
+    } else {
+      _timer = Timer(const Duration(milliseconds: 200), () {
+        if (!_lineFeedCall) {
```

Comment:
If we're going to do this with some timer waiting for barcode to end, I don't think mixing it with existing idea is either transparent to the end user or easier to maintain.
Both, problem with extra function call and this can be handled with public property available for user to set it and decide which way user wants this library to work. 
Default would be AS IS, wait for linefeed, alternative would be "I don't care about linefeed, let's use timer"

Reply:
dude,you are right,there's a new public property available for timer  in this commit